File: leetcode.com/1310_xor_queries_of_a_subarray/solution.go
import "fmt"

func xorQueries(arr []int, queries [][]int) []int {
    res := []int{}

    for i := 0 ; i < len(queries) ; i++ {
        v := 0

        for j := queries[i][0] ; j <= queries[i][1] ; j++ {
            v ^= arr[j]
        }

        res = append(res, v)
    }

    return res
}

File: leetcode.com/1310_xor_queries_of_a_subarray/solution.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        s, prefix = 0, []
        for n in arr:
            s ^= n
            prefix.append(s)

        res = []
        for left, right in queries:

            if left == 0:
                v = prefix[right]
            else:
                v = prefix[right] ^ prefix[left - 1]

            res.append(v)

        return res
